internal/pkg/db/redis/data.go

```diff
@@ -780,6 +780,12 @@ func addEvent(conn redis.Conn, e contract.Event) (id string, err error) {
 		return "", err
 	}
 
+	for i, reading := range e.Readings {
+		if reading.Device == "" {
+			e.Readings[i].Device = e.Device
+		}
+	}
```

Comment:
This one works because Redis is operating on the contract model directly and not mapping it to a provider specific model.